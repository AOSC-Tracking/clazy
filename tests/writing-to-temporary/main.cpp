struct Foo
{
    int m;
    void setM(int v) { m = v; }
    int setX(int v) { return 0; }
    void setY() const {}
    static void setM_static(int) {}
};


Foo getFoo() { return Foo(); }
Foo& getFooRef() { static Foo f; return f; }
Foo* getFooPtr() { return new Foo(); }

void test()
{
    getFoo().setM(1); // Warning
    getFoo().setM_static(1);
    getFoo().setX(1); // OK
    getFoo().setY(); // OK
    getFooRef().setM(1); // OK
    getFooPtr()->setM(1); // OK

    Foo f;
    f.setM(1); // OK
}

#include <QtCore/QSize>

QSize getSize() { return QSize(); }

void testKnownTypes()
{
    getSize().transpose(); // Warning
}

#include <QtXml/QDomNode>

QDomNode getNode() { return {}; }

void testDisallowedType() // bug #354363
{
    getNode().firstChild(); // OK
    QDomNode node;
    node.firstChild().setPrefix(""); // OK
}
