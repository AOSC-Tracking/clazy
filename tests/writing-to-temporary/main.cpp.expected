main.cpp:34:5: warning: Call to temporary is a no-op: QSize::transpose [-Wclazy-writing-to-temporary]
